src/main/java/com/savvato/tribeapp/services/PhraseServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.savvato.tribeapp.services;
+
+import com.savvato.tribeapp.dto.PhraseDTO;
+import com.savvato.tribeapp.entities.Phrase;
+import com.savvato.tribeapp.repositories.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class PhraseServiceImpl implements PhraseService {
+
+    @Autowired
+    UserPhraseService userPhraseService;
+
+    @Autowired
+    PhraseRepository phraseRepository;
+
+    @Autowired
+    AdverbRepository adverbRepository;
+
+    @Autowired
+    NounRepository nounRepository;
+
+    @Autowired
+    PrepositionRepository prepositionRepository;
+
+    @Autowired
+    VerbRepository verbRepository;
+
+    @Override
+    public Optional<List<PhraseDTO>> getListOfPhraseDTOByUserId(Long userId) {
+
+        // create list of PhraseDtos
+        List<PhraseDTO> phraseDTOS = new ArrayList<>();
+
+        // Create list of phrase ids
+        Optional<List<Long>> optPhraseIds = userPhraseService.findPhraseIdsByUserId(userId);
+
+        // Create list of all phrases from those phrase ids
+        if (optPhraseIds.isPresent()) {
+            List<Long> phraseIds = optPhraseIds.get();
+            List<Phrase> phrases = new ArrayList<>();
+
+            // loop through list of phrase ids and add each phrase to list
+            for (Long phraseId : phraseIds) {
+                Optional<Phrase> optPhrase = phraseRepository.findPhraseByPhraseId(phraseId);
+                if (optPhrase.isPresent()) {
+                    phrases.add(optPhrase.get());
+                }
+            }
+
+            // loop through phrases and get words
+            for (Phrase phrase : phrases) { //do we need to check again if empty
```

Comment:
You could check if empty, here but I don't think it adds a lot of value.. A `for` statement with an empty list is a cheap operation. Preventing it doesn't save us processing time as much as it adds a bit more code.

Reply:
Thanks for confirming.